vllm/v1/structured_output/__init__.py

```diff
@@ -40,6 +40,14 @@ def __init__(self, vllm_config: VllmConfig):
         self._grammar_bitmask: Optional[torch.Tensor] = None
         self._full_mask = torch.tensor(-1, dtype=torch.int32)
 
+        max_batch_size = self.vllm_config.scheduler_config.max_num_seqs
+        self.fill_bitmask_parallel_threshold = 128
+        if self.fill_bitmask_parallel_threshold < max_batch_size:
+            self.fill_bitmask_parallel_batch_size = 16
+            max_workers = max(1, min(multiprocessing.cpu_count(), 8))
```

Comment:
Nit: I'd never want to use all logical cores for this, maybe half?

Reply:
I'll push a change for this

Reply:
maintainer write access is disabled on this fork, so @benchislett you'll have to make this change